data/bookkeeping/block_test.go

```diff
@@ -769,3 +770,107 @@ func TestNextRewardsRateWithFixNextRewardLevelOverflow(t *testing.T) {
+	crypto.RandBytes(block.BlockHeader.GenesisHash[:])
+
+	for i := 0; i < 1024; i++ {
+		txn := transactions.Transaction{
+			Type: protocol.PaymentTx,
+			Header: transactions.Header{
+				GenesisHash: block.BlockHeader.GenesisHash,
+			},
+			PaymentTxnFields: transactions.PaymentTxnFields{
+				Amount: basics.MicroAlgos{Raw: crypto.RandUint64()},
+			},
+		}
+
+		crypto.RandBytes(txn.Sender[:])
+		crypto.RandBytes(txn.PaymentTxnFields.Receiver[:])
+
+		sigtxn := transactions.SignedTxn{Txn: txn}
+		ad := transactions.ApplyData{}
+
+		stib, err := block.BlockHeader.EncodeSignedTxn(sigtxn, ad)
+		a.NoError(err)
+
+		block.Payset = append(block.Payset, stib)
+	}
+
+	tree, err := block.TxnMerkleTree()
+	a.NoError(err)
+	rootSliceSHA512_256 := tree.Root()
+
+	tree, err = block.TxnMerkleTreeSHA256()
+	a.NoError(err)
+	rootSliceSHA256 := tree.Root()
+
+	badDigestSlice := []byte("(>^-^)>")
+
+	/* Test V32 */
+	a.False(block.ContentsMatchHeader())
+
+	copy(block.BlockHeader.TxnRoot.DigestSha512_256[:], rootSliceSHA512_256)
+	block.BlockHeader.TxnRoot.DigestSha256 = crypto.Digest{}
+	a.True(block.ContentsMatchHeader())
+
+	copy(block.BlockHeader.TxnRoot.DigestSha512_256[:], rootSliceSHA512_256)
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], rootSliceSHA256)
+	a.False(block.ContentsMatchHeader())
+
+	copy(block.BlockHeader.TxnRoot.DigestSha512_256[:], badDigestSlice)
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], rootSliceSHA256)
+	a.False(block.ContentsMatchHeader())
+
+	block.BlockHeader.TxnRoot.DigestSha512_256 = crypto.Digest{}
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], rootSliceSHA256)
+	a.False(block.ContentsMatchHeader())
+
+	/* Test Consensus Future */
+	// Create a block with SHA256 TxnRoot
+	block.CurrentProtocol = protocol.ConsensusFuture
+
+	block.BlockHeader.TxnRoot.DigestSha512_256 = crypto.Digest{}
+	block.BlockHeader.TxnRoot.DigestSha256 = crypto.Digest{}
+	a.False(block.ContentsMatchHeader())
+
+	// Now update the SHA256 header to its correct value
+	copy(block.BlockHeader.TxnRoot.DigestSha512_256[:], rootSliceSHA512_256)
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], rootSliceSHA256)
+	a.True(block.ContentsMatchHeader())
+
+	copy(block.BlockHeader.TxnRoot.DigestSha512_256[:], badDigestSlice)
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], rootSliceSHA256)
+	a.False(block.ContentsMatchHeader())
+
+	copy(block.BlockHeader.TxnRoot.DigestSha512_256[:], rootSliceSHA512_256)
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], badDigestSlice)
+	a.False(block.ContentsMatchHeader())
+
+	block.BlockHeader.TxnRoot.DigestSha512_256 = crypto.Digest{}
+	copy(block.BlockHeader.TxnRoot.DigestSha256[:], rootSliceSHA256)
+	a.False(block.ContentsMatchHeader())
+}
+
+func TestBlockHeader_Serialization(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	a := require.New(t)
+
+	// This serialized block header was generated from V32 e2e test, using the old BlockHeader struct which contains only TxnRoot SHA512_256 value
```

Comment:
could you maybe add the commands used to obtain this value?

Reply:
I checked out the master branch and added encode + print in the middle of an e2e test to get a somewhat full `BlockHeader`.
If it breaks you can alway checkout the previous commit and look at the actual decoded struct, right?

Reply:
yes, I understand. Just add couple pointers, like:
1. edit XYZ test
2. add the following (example)
```
blk, err := ledger.Block(10)
require.NoError(t, err)
fmt.Printf("%x\n", protocol.Encode(&blk))
```
This is good for the tests maintainability